### PR TITLE
Choropleth API improvements

### PR DESCRIPTION
**Pull request by alexbaden**
# Merge Checklist
## :wrench: Issue(s) fixed:
- [ ] Author referenced issue(s) fixed by this PR:
- [ ] Fixes #0

## :smoking: Smoke Test
- [ ] Works in chrome
- [ ] Works in firefox
- [ ] Works in safari
- [ ] Works in ie edge
- [ ] Works in ie 11

## :ship: Merge
- [ ] author crafted PR's title into release-worthy commit message.

### CODE REVIEW

File: src/mixins/raster-layer-poly-mixin.js [hunk 1]
```diff
@@ -180,7 +186,9 @@ export default function rasterLayerPolyMixin(_layer) {
   }
 
   function usesAutoColors() {
-    return state.encoding.color.domain === "auto"
+    return state.encoding.color.domain === undefined
```
Comment (jrajav) [inline src/mixins/raster-layer-poly-mixin.js]: I'm confused why this change is necessary? state.encoding.color.domain === "auto" will be false if state.encoding.color.domain is undefined, anyway.

File: dist/mapdc.js [hunk 2]
```diff
@@ -26631,21 +26629,24 @@ function rasterLayerPolyMixin(_layer) {
         layerFilter = _ref$layerFilter === undefined ? [] : _ref$layerFilter,
         filtersInverse = _ref.filtersInverse;
 
-    var selfJoin = state.data[0].table === state.data[1].table;
+    var doJoin = state.data.length > 1;
+    var hasColorMeasure = !(state.encoding.color.domain === undefined);
 
-    var groupby = {
+    var groupby = hasColorMeasure && !(state.data[0].attr === "rowid") ? {
```
Comment (biovisualize) [inline dist/mapdc.js]: Maybe `!(state.data[0].attr === "rowid")` could go in a variable to give some info about what it means

File: dist/mapdc.js [hunk 3]
```diff
@@ -26782,15 +26780,31 @@ function rasterLayerPolyMixin(_layer) {
...
-      domain: autocolors ? { data: getStatsLayerName(), fields: ["mincolor", "maxcolor"] } : state.encoding.color.domain,
-      range: colorRange,
-      nullValue: "rgba(214, 215, 214, 0.65)",
-      default: "rgba(214, 215, 214, 0.65)"
-    }];
+    var scales = [];
+    var fillColor = {};
+    var useColorScale = state.encoding.color.value === undefined;
+    if (useColorScale) {
+      var colorRange = state.encoding.color.range.map(function (c) {
```
Comment (biovisualize) [inline dist/mapdc.js]: I guess we are confident that `state.encoding.color.range` will always be populated and will be an array.

File: dist/mapdc.js [hunk 4]
```diff
@@ -26782,15 +26780,31 @@ function rasterLayerPolyMixin(_layer) {
...
+        return (0, _utilsVega.adjustOpacity)(c, state.encoding.color.opacity);
```
Comment (biovisualize) [inline dist/mapdc.js]: I'm not sure I understand how this line works. I guess it's equivalent to `_utilsVega.adjustOpacity(c, state.encoding.color.opacity)`, but what is the purpose of the first `0`?

File: dist/mapdc.js [hunk 5]
```diff
@@ -26839,7 +26850,7 @@ function rasterLayerPolyMixin(_layer) {
   _layer._genVega = function (chart, layerName, group, query) {
     _vega = _layer.__genVega({
       layerName: layerName,
-      filter: _layer.crossfilter().getFilterString(_layer.dimension().getDimensionIndex()),
+      filter: _layer.crossfilter().getFilterString(_layer.dimension() ? _layer.dimension().getDimensionIndex() : null),
```
Comment (biovisualize) [inline dist/mapdc.js]: Maybe the ternary operator could be extracted in a variable

File: src/mixins/raster-layer-poly-mixin.js (hunk 1 above)
Comment (alexbaden) [reply src/mixins/raster-layer-poly-mixin.js]: Good catch. Likely a holdover from when `state.encoding.color` could be undefined. I removed it.

File: src/mixins/raster-layer-poly-mixin.js [hunk 6]
```diff
@@ -253,20 +258,36 @@ export default function rasterLayerPolyMixin(_layer) {
...
     }
 
-    const colorScaleName = getColorScaleName(layerName)
-    const scales = [
-      {
+    let scales = []
+    let fillColor = {}
+    const useColorScale = state.encoding.color.value === undefined
+    if (useColorScale) {
+      const colorRange = state.encoding.color.range.map(c =>
```
Comment (biovisualize) [inline src/mixins/raster-layer-poly-mixin.js]: We could clean-up to be consistent at some point between using `function() {}` or the fat arrow for anonymous functions

File: dist/mapdc.js (hunk 4 above)
Comment (biovisualize) [reply dist/mapdc.js]: Ah! @jrajav made me realize that I was reviewing the compiled file, which adds these things. So @alexbaden feel free to ignore everything that doesn't relate to an actual code change :)

**Comment by tmostak** [comment]
Hi @alexbaden should we remove mapd_geo from the dimension list now?
